src/Pages/EditTicketForm.jsx

```diff
@@ -0,0 +1,148 @@
+import { useState } from "react";
+
+
+function EditTicketForm({ competitions, ticket }) {
```

Comment:
I promised you an example of how this would look as a controlled form, so here we go.

Right now this component is uncontrolled because it as its own state (`const [firstName, setFirstName] = useState(ticket.firstName);` etc.)

For a controlled form (which is preferable), all of that state would be provided as props. So you'd end up with

```
function EditTicketForm({competitions, tickets, firstName, setFirstName, lastName, setLastName, ...etc})
```

That means that the form would no longer be holding its own state and would be "controlled" by the parent component. It's much easier to debug and manage react apps that don't have uncontrolled forms in them.